shoebot/__init__.py

```diff
@@ -378,7 +378,7 @@ def run(
     elif background_thread:
         try:
```

Comment:
Can you try adding `sleep(0.01)` on the line before the `while` and see if it makes any difference ?

The code in this PR definitely improves things - even if our code is possibly hiding some bugs underneath.